pykafka/simpleconsumer.py

```diff
@@ -520,25 +521,16 @@ def _handle_success(parts):
 
                 time.sleep(i * (self._offsets_channel_backoff_ms / 1000))
 
-                if 0 in parts_by_error:
-                    parts_by_error.pop(0)
-                errored_partitions = dict(
-                    (part, owned_partition_offsets[part])
-                    for errcode, parts in parts_by_error.iteritems()
-                    for part, _ in parts)
-                owned_partition_offsets = dict(errored_partitions.items() +
-                                               owned_partition_offsets.items())
-
-            for errcode, owned_partitions in parts_by_error.iteritems():
-                if errcode != 0:
-                    for owned_partition in owned_partitions:
-                        owned_partition.fetch_lock.release()
+                for errcode, owned_partitions in parts_by_error.iteritems():
```

Comment:
This is a good catch, though I'm surprised it wasn't already fixed. I could've sworn I made this exact change somewhere else.